pkg/app/app.go

```diff
@@ -536,6 +536,10 @@ func (h HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		log.Println("Error: ", err)
 		var e *apperr.AppError
 		if errors.As(err, &e) {
+			if e.Error() == accounts.MissingUsername {
+				http.Redirect(w, r, "/username", http.StatusSeeOther)
```

Comment:
And can we append a kind of 'callback' url as suffix to move back to current page after login?

Reply:
We can. How about redirecting back to the original page when `HandleUsernameOnlyLogging` gets a valid username from the POST method?